morf-integration-test/src/test/java/org/alfasoftware/morf/integration/TestDatabaseUpgradeIntegration.java

```diff
@@ -937,6 +941,56 @@ private void doAddColumn(Class<? extends UpgradeStep> upgradeStep) throws SQLExc
   }
 
 
+  /**
+   * Tests adding non-nullable column to the table,
+   * and leaving a default value behind.
+   */
+  @Test
+  public void testAddColumnWithoutDropDefault1() throws SQLException {
+    Schema expected = replaceTablesInSchema(
+      table("WithDefaultValue")
+        .columns(
+          column("id", DataType.STRING, 3).primaryKey(),
+          column("version", DataType.INTEGER).defaultValue("0"),
+          column("anotherValue", DataType.STRING, 10)
+        )
+      );
+
+    try {
+      verifyUpgrade(expected, AddColumnWithoutDropDefaultValue.class);
+
+      fail ("Should not upgrade because of forgotten DEFAULT value");
+    }
+    catch (NoUpgradePathExistsException e) { /* happy path */ }
+  }
+
+
+  /**
+   * Tests adding non-nullable column to the table,
+   * and leaving a default value behind.
+   */
+  @Test
+  public void testAddColumnWithoutDropDefault2() throws SQLException {

```

Comment:
What's the difference between these tests, I see the extra `defaultValue("OLD")` - is that "WithForgottenDefaultValue" ?

Reply:
testAddColumnWithoutDropDefault1:
- upgrade stops with NoUpgradePathExistsException
- all as expected

testAddColumnWithoutDropDefault2:
- upgrade proceeds and even succeeds, and only then things fail on schema comparison
- this shows how users can "cheat" the upgrade path finder into upgrading
- this actually leaves behind the lurking default value if I am not mistaken

Perhaps some renaming would be nice?